app/src/main/java/me/ccrama/redditslide/Vote.java

```diff
@@ -13,19 +13,21 @@
 import net.dean.jraw.models.PublicContribution;
 import net.dean.jraw.models.VoteDirection;
 
+import java.lang.ref.WeakReference;
+
 /**
  * Created by ccrama on 9/19/2015.
  */
 public class Vote extends AsyncTask<PublicContribution, Void, Void> {
 
     private final VoteDirection direction;
-    private View v;
-    private Context c;
+    private WeakReference<View> v;
+    private WeakReference<Context> c;
```

Comment:
Thanks for the PR @hassan963 

With the adoption of weak refs in this sort of case, what happens to the NPEs if .get() returns null, does it cause a crash?

Reply:
Yes, it will crash. I have checked if the ref is null or not in Vote.java class, so it wont cause any NPEs. But by mistake I did not check on other classes. Below classes
```suggestion
1. me.ccrama.redditslide.Services.CommentScreenTask.java
2. me.ccrama.redditslide.Services.InboxTask.java
3. me.ccrama.redditslide.Services.LiveThreadTask.java
```